syntax = "proto2";

package janitor;

message PackageMetadata {
   optional string name = 1;

   optional string maintainer_email = 2;

   repeated string uploader_email = 3;

   optional int32 insts = 4;

   optional string vcs_type = 5;
   optional string vcs_url = 6;
   optional string vcs_browser = 7;

   optional string commit_id = 8;
   optional string vcswatch_status = 9;
   optional string archive_version = 10;
   optional string vcswatch_version = 11;

   optional bool in_base = 13;

   optional bool removed = 12 [default=false];
};

message PackageRemoval {
   optional string name = 1;
   optional string version = 2;
};

message PackageList {
    repeated PackageMetadata package = 1;
    repeated PackageRemoval removal = 2;
};
